storage/replica_raftstorage.go

```diff
@@ -638,7 +651,8 @@ func (r *Replica) applySnapshot(
 	// unreplicated keys from the snapshot.
 	unreplicatedPrefix := keys.MakeRangeIDUnreplicatedPrefix(desc.RangeID)
 
-	// Write the snapshot into the range.
+	// Write the snapshot into the range. The KVs and log entries are all written
```

Comment:
Maybe "The KVs and log entries are all written to distinct keys so we can use a distinct batch" should go up where we create the distinct batch above.